component/authenticator/src/main/java/org/wso2/carbon/identity/application/authenticator/totp/util/TOTPUtil.java

```diff
@@ -543,4 +547,83 @@ public static String getEnableTOTPPageFromXMLFile(AuthenticationContext context,
 		}
 		return enableTOTPPage;
 	}
-}
\ No newline at end of file
+
+	/**
+	 * Get Account Lock Connector Configs.
+	 *
+	 * @param tenantDomain Tenant domain.
+	 * @return Account Lock Connector Configs.
+	 * @throws AuthenticationFailedException Exception on authentication failure.
+	 */
+	public static Property[] getAccountLockConnectorConfigs(String tenantDomain) throws AuthenticationFailedException {
+
+		Property[] connectorConfigs;
+		try {
+			connectorConfigs = TOTPDataHolder.getInstance()
+					.getIdentityGovernanceService()
+					.getConfiguration(
+							new String[]{
+									TOTPAuthenticatorConstants.PROPERTY_ACCOUNT_LOCK_ON_FAILURE,
+									TOTPAuthenticatorConstants.PROPERTY_ACCOUNT_LOCK_ON_FAILURE_MAX,
+									TOTPAuthenticatorConstants.PROPERTY_ACCOUNT_LOCK_TIME,
+									TOTPAuthenticatorConstants.PROPERTY_LOGIN_FAIL_TIMEOUT_RATIO
+							}, tenantDomain);
+		} catch (IdentityGovernanceException e) {
+			throw new AuthenticationFailedException(
+					"Error occurred while retrieving account lock connector configuration", e);
+		}
+		return connectorConfigs;
+	}
+
+	/**
+	 * Check whether account locking is enabled for TOTP.
+	 *
+	 * @return True if account locking is enabled for TOTP.
+	 */
+	public static boolean isAccountLockingEnabledForTotp() {
+
+		return Boolean.parseBoolean(
+				getTOTPParameters().get(TOTPAuthenticatorConstants.ENABLE_ACCOUNT_LOCKING_FOR_FAILED_ATTEMPTS));
+	}
+
+	/**
+	 * Check whether the user account is locked.
+	 *
+	 * @param userName        The username of the user.
+	 * @param tenantDomain    The tenant domain.
+	 * @param userStoreDomain The userstore domain.
+	 * @return True if the account is locked.
+	 * @throws AuthenticationFailedException Exception on authentication failure.
+	 */
+	public static boolean isAccountLocked(String userName, String tenantDomain, String userStoreDomain)
+			throws AuthenticationFailedException {
+
+		try {
+			return TOTPDataHolder.getInstance().getAccountLockService()
+					.isAccountLocked(userName, tenantDomain, userStoreDomain);
+		} catch (AccountLockServiceException e) {
+			throw new AuthenticationFailedException(
+					String.format("Error while validating account lock status of user: %s.", userName), e);
+		}
+	}
+
+	/**
+	 * Check whether the user being authenticated via a local authenticator or not.
+	 *
+	 * @param context Authentication context.
+	 * @return Whether the user being authenticated via a local authenticator.
+	 */
+	public static boolean isLocalUser(AuthenticationContext context) {
+
+		Map<Integer, StepConfig> stepConfigMap = context.getSequenceConfig().getStepMap();
+		if (stepConfigMap != null) {
```

Comment:
Shall we modify the method to end when by changing the condition?

Reply:
Fixed in 8307785